### PR TITLE
feat: Lesson 7 - Building Chat Applications - TypeScript Sample

### PR DESCRIPTION
**Pull request by glaucia86**
This PR is a TypeScript version of Lesson 07: **[Building Generative AI-Powered Chat Applications](https://microsoft.github.io/generative-ai-for-beginners/#/07-building-chat-applications/README?wt.mc_id=academic-105485-koreyst&id=building-generative-ai-powered-chat-applications)**

## How to run the example?

1. Create a `.env` file at the root of the application and include the information for `AZURE_OPENAI_ENDPOINT` and `AZURE_OPENAI_KEY`. There is a file named `.env-sample` explaining how to create the `.env` file.

2. To install dependencies, navigate to

> cd: `typescript/chat-completions-app

Then execute the command:

> `npm install`

3. After the necessary dependencies are installed in the application, open the terminal and run the command:

> `npm run build`

4. Finally, execute the command:

> `npm run start`

5. You will have the application running perfectly.

You can see in action in this gif below:

chat-completion-lesson-07

### CODE REVIEW

File: 07-building-chat-applications/typescript/chat-completions-app/package.json [hunk 1]
```diff
@@ -0,0 +1,39 @@
+{
+  "name": "chat-completions-app",
+  "version": "1.0.0",
+  "description": "a simple application how to use the chat completions with Azure OpenAI in TypeScript",
+  "main": "main.js",
+  "scripts": {
+    "build": "rimraf build && tsc",
+    "preserve": "npm run build",
+    "start": "npm run build && cross-env NODE_ENV=development concurrently \"nodemon\"",
```
Comment (manekinekko) [inline 07-building-chat-applications/typescript/chat-completions-app/package.json:9]: the start command can be simplified to this (I don't this need cross-env and concurrently packages):
```suggestion
    "start": "nodemon",
```

File: 07-building-chat-applications/typescript/chat-completions-app/package.json [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  "keywords": [
+    "typescript",
+    "azure",
+    "openai"
+  ],
+  "author": "",
+  "license": "MIT",
+  "devDependencies": {
+    "@types/node": "^20.10.8",
+    "concurrently": "^8.2.2",
```
Comment (manekinekko) [inline 07-building-chat-applications/typescript/chat-completions-app/package.json:21]:
```suggestion
```

File: 07-building-chat-applications/typescript/chat-completions-app/package.json [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    "nodemon": "^3.0.2",
+    "ts-node": "^10.9.2",
+    "typescript": "^5.3.3"
+  },
+  "dependencies": {
+    "@azure/openai": "^1.0.0-beta.10",
+    "cross-env": "^7.0.3",
```
Comment (manekinekko) [inline 07-building-chat-applications/typescript/chat-completions-app/package.json:28]:
```suggestion
```

Comment (glaucia86) [reply 07-building-chat-applications/typescript/chat-completions-app/package.json]: Thanks a lot for this, @manekinekko. I included all your suggestions in the PR :)

**Review by koreyspace — APPROVED** [review]

**Pull request merged**